src/pydicom/dataset.py

```diff
@@ -2109,16 +2112,175 @@ def overlay_array(self, group: int) -> "numpy.ndarray":
+            the installation of additional packages to perform the actual pixel
+            data decoding. See the :doc:`pixel data decompression documentation
+            </guides/user/image_data_handlers>` for more information.
+
+        .. versionchanged:: 3.0
+
+            The backend used for pixel data decoding has changed from the
+            :mod:`~pydicom.pixel_data_handlers` module to the
+            :mod:`~pydicom.pixels` module. The behavior of the new backend
+            is not backwards compatible with the old one, in particular the
+            default color space should now be RGB when previously YCbCr data
+            was returned.
+
+            To revert to the deprecated :mod:`~pydicom.pixel_data_handlers`
+            backend pass ``use_v2_backend=True`` to the
+            :meth:`~pydicom.dataset.Dataset.pixel_array_options` method::
+
+                >>> from pydicom import examples
+                >>> ds = examples.ct
+                >>> ds.pixel_array_options(use_v2_backend=True)
+                >>> arr = ds.pixel_array
+
+            The :mod:`~pydicom.pixel_data_handlers` module and the
+            `use_v2_backend` keyword argument will be removed in v4.0.
+
         Returns
         -------
         numpy.ndarray
-            The (7FE0,0008) *Float Pixel Data*, (7FE0,0009) *Double Float
-            Pixel Data* or (7FE0,0010) *Pixel Data* converted to a
-            :class:`numpy.ndarray`.
+            The contents of the (7FE0,0008) *Float Pixel Data*, (7FE0,0009)
+            *Double Float Pixel Data* or (7FE0,0010) *Pixel Data* elements
+            converted to a :class:`numpy.ndarray`. The array will be shaped as:
+
+            * (rows, columns) for single frame, single sample data
+            * (rows, columns, samples) for single frame, multi-sample data
+            * (frames, rows, columns) for multi-frame, single sample data
+            * (frames, rows, columns, samples) for multi-frame, multi-sample data
+
+            When using the :mod:`pydicom.pixels` backend the decoding options
+            used with the returned array can be customized via the
+            :meth:`~pydicom.dataset.Dataset.pixel_array_options` method.
+
+        See Also
+        --------
+        pydicom.pixels.pixel_array
+            A function for returning the pixel data from the path to a dataset,
+            a readable file-like containing a dataset or a
+            :class:`~pydicom.dataset.Dataset` instance. Can be used to minimize
+            the memory required to return the pixel data when used with a path
+            or file-like.
+        pydicom.pixels.iter_pixels
+            Similar to :func:`pydicom.pixels.pixel_array` but returns a generator
+            that iterates through the image frames.
         """
         self.convert_pixel_data()
         return cast("numpy.ndarray", self._pixel_array)
 
+    def pixel_array_options(
```

Comment:
Would have expected `set_pixel_array_options`, as this is a setter.

Reply:
It's already a pretty long name, though. I think I'll leave it?

Reply:
I can live with that :)